crates/matrix-sdk/tests/integration/event_cache.rs

```diff
@@ -1543,6 +1549,145 @@ async fn test_apply_redaction_when_redaction_comes_later() {
+    let event_cache = client.event_cache();
+    event_cache.subscribe().unwrap();
+    event_cache.enable_storage().unwrap();
+
+    let room = mock_server.sync_joined_room(&client, room_id).await;
+    let (room_event_cache, _room_event_cache_drop_handle) = room.event_cache().await.unwrap();
+
+    let (initial_updates, mut updates_stream) = room_event_cache.subscribe().await;
+
+    // Initial events!
+    //
+    // Only 1 events is loaded, as expected, from the last chunk.
+    {
+        assert_eq!(initial_updates.len(), 1);
+        assert_event_id!(initial_updates[0], "$ev1");
+
+        // The stream of updates is waiting, patiently.
+        assert!(updates_stream.is_empty());
+    }
+
+    // Sync a redaction for `$ev0`.
+    mock_server
+        .sync_room(
+            &client,
+            JoinedRoomBuilder::new(room_id).add_timeline_event(
+                event_factory
+                    .redaction(event_id!("$ev0"))
+                    .event_id(event_id!("$ev2"))
+                    .into_raw_sync(),
+            ),
+        )
+        .await;
+
+    // Let's check the stream.
+    let update = updates_stream.recv().await.unwrap();
+
+    assert_matches!(update, RoomEventCacheUpdate::UpdateTimelineEvents { diffs, .. } => {
+        // 1 diff for the `m.room.redaction`. The event being redacted is not
+        // in-memory yet, it's only in the store, so no update for it.
+        assert_eq!(diffs.len(), 1);
+
+        assert_matches!(&diffs[0], VectorDiff::Append { values: events } => {
+            assert_eq!(events.len(), 1);
+            assert_event_id!(&events[0], "$ev2");
+        });
+    });
+
+    assert!(updates_stream.is_empty());
+
+    // To see if `$ev0` has been redacted in the store, let's paginate!
+    let outcome = room_event_cache.pagination().run_backwards_until(1).await.unwrap();
+
+    // 1 event, no surprise.
+    assert_eq!(outcome.events.len(), 1);
+    assert_event_id!(outcome.events[0], "$ev0");
+    assert_matches!(
+        outcome.events[0].raw().deserialize().unwrap(),
+        AnySyncTimelineEvent::MessageLike(AnySyncMessageLikeEvent::RoomMessage(event)) => {
+            // The event has been redacted!
+            assert!(event.as_original().is_none());
+        }
+    );
+
+    // Let's check the stream. It should reflect what the `pagination_outcome`
+    // provides.
+    let update = updates_stream.recv().await.unwrap();
+
+    assert_matches!(update, RoomEventCacheUpdate::UpdateTimelineEvents { diffs, .. } => {
+        assert_eq!(diffs.len(), 1);
+
+        assert_matches!(&diffs[0], VectorDiff::Insert { index: 0, value: event } => {
+            assert_event_id!(event, "$ev0");
+            assert_matches!(
+                event.raw().deserialize().unwrap(),
+                AnySyncTimelineEvent::MessageLike(AnySyncMessageLikeEvent::RoomMessage(event)) => {
+                    // The event has been redacted!
+                    assert!(event.as_original().is_none());
```

Comment:
```suggestion
                AnySyncTimelineEvent::MessageLike(event) => {
                    // The event has been redacted!
                    assert!(event.is_redacted());
```

Not quite the same, but if we only care about the fact that the event is redacted might be nicer to read.

Reply:
Cool, didn't know about this one! Fixed up and rebased.